rotkehlchen/chain/ethereum/decoding/decoder.py

```diff
@@ -254,22 +254,40 @@ def decode_transaction_hashes(self, ignore_cache: bool, tx_hashes: List[EVMTxHas
         self.reload_from_db()
         tx_module = EthTransactions(ethereum=self.ethereum_manager, database=self.database)
 
-        for tx_hash in tx_hashes:
-            try:
-                receipt = tx_module.get_or_query_transaction_receipt(tx_hash)
-            except RemoteError as e:
-                raise InputError(f'Hash {tx_hash.hex()} does not correspond to a transaction') from e  # noqa: E501
-
-            # TODO: Change this if transaction filter query can accept multiple hashes
-            txs = self.dbethtx.get_ethereum_transactions(
-                filter_=ETHTransactionsFilterQuery.make(tx_hash=tx_hash),
-                has_premium=True,  # ignore limiting here
+        # If no transaction hashes are passed, decode all transactions.
+        if len(tx_hashes) == 0:
```

Comment:
Please make it optional and change types everywhere to showcase it is optional. It's much more explicit than using an empty list.